rfcs/convergence/async-import-hidden-components.md

```diff
@@ -0,0 +1,88 @@
+# RFC: Asynchronously import hidden components
+
+Driver: @czearing
+
+(Background provided by @dzearing)
+
+## Summary
+
+When components such as `Tooltip` and `ContextualMenu` are not present in the DOM, they should be asynchronously imported to avoid bundle size related issues.
+
+## Background
+
+In the past we've avoided the use of async imports to prescribe split points, because the technology was new, and bundlers didn't have strong support for it. We built a proprietary webpack plugin for managing re-aliasing. This was done before webpack supported async import split points, so that we could ensure the right files were aliased, rather than asking each partner to set this up. But (I believe) the only partner to take advantage of this plugin was OOUI. (Was very proprietary, not well documented, etc.) But now with async imports, we don't need plugins. With the new components being worked on and tools being modernized, we should revisit this issue.
+
+If you don't know what they are; it is a way to asynchronously import source code. Bundlers use async imports as a cue for creating a split point in the bundle.
+
+Read more about how React itself helps make this simple for components: https://reactjs.org/docs/code-splitting.html
+
+## Problem statement
+
+We have several scenarios where async imports would be beneficial for minimizing bundle size impact on our partners apps, especially for initial page load.
```

Comment:
Users with limited bandwidth or where the last mile is distant from origin, the cost of each JS download can be mostly the request resolution and server connection.  Once the data starts downloading, the payload size doesn't cost as much relative to the connection.  In other words, if there are lots of small requests it can be more expensive.  

This is typically solved with CDNs or edge caching. 
- What the experience would be with devs wanting to use a CDN? 
- Would each dynamic import be able to be resolved to a CDN address?  
- Does having lots of code splits mean that there are more things being invalidated as fluent changes?

Reply:
[SplitChunksPlugin](https://webpack.js.org/plugins/split-chunks-plugin/) has options to help configure this on the consumer side, including without any knowledge of paths/filenames. Most nontrivial apps will already have async imports and multiple bundles, and probably have SplitChunksPlugin or a similar solution set up to customize it. So I don't think this is something we need to worry about directly.